makeflow/src/makeflow_wrapper.c

```diff
@@ -0,0 +1,274 @@
+    list_push_tail(w->output_files, f);
+}
+
+struct list *makeflow_wrapper_generate_files( struct list *result, struct list *input, struct dag_node *n, struct makeflow_wrapper *w)
+{
+    char *f;
+    char *nodeid = string_format("%d",n->nodeid);
+
+    struct list *files = list_create();
+
+    list_first_item(input);
+    while((f = list_next_item(input)))
+    {
+        char *filename = strdup(f);
+        filename = string_replace_percents(filename, nodeid);
+
+		char *remote, *p;
+	    char *f = strdup(filename);
+		struct dag_file *file;
+		free(filename);
+		p = strchr(f, '=');
+		if(p) {
+			*p = 0;
+			filename = xxstrdup(f);
+			remote = xxstrdup(p+1);
+			*p = '=';
+			file = dag_file_lookup_or_create(n->d, filename);
+			if(!n->local_job){
+				itable_insert(w->remote_names, (uintptr_t) file, remote);
+				hash_table_insert(w->remote_names_inv, remote, (void *)file);
+			}
+		} else {
+			filename = xxstrdup(f);
+			remote = NULL;
+			file = dag_file_lookup_or_create(n->d, filename);
+		}
+		free(f);
+        list_push_tail(files, file);
+    }
+    free(nodeid);
+
+	result = list_splice(result, files);
+
+    return result;
+}
+
+/* Returns the remotename used in wrapper for local name filename */
+const char *makeflow_wrapper_get_remote_name(struct makeflow_wrapper *w, struct dag *d, const char *filename)
+{
+	struct dag_file *f;
+	char *name;
+
+	f = dag_file_from_name(d, filename);
+	name = (char *) itable_lookup(w->remote_names, (uintptr_t) f);
+
+	return name;
+}
+
+/*
+ * Prepare for monitoring by creating wrapper command and attaching the
+ * appropriate input and output dependencies.
+ * */
+void makeflow_prepare_for_monitoring( struct makeflow_monitor *m, char *log_dir, char *log_format)
+{
+    m->log_prefix = string_format("%s/%s", log_dir, log_format);
+    char *log_name;
+
+	if(m->exe_remote){
+		log_name = string_format("%s=%s", m->exe, m->exe_remote);
+		makeflow_wrapper_add_input_file(m->wrapper, log_name);
+		free(log_name);
+	} else {
+		makeflow_wrapper_add_input_file(m->wrapper, m->exe);
+	}
+
+    log_name = string_format("%s.summary", m->log_prefix);
+    makeflow_wrapper_add_output_file(m->wrapper, log_name);
+    free(log_name);
+
+    if(m->enable_time_series)
+    {
+        log_name = string_format("%s.series", m->log_prefix);
+		makeflow_wrapper_add_output_file(m->wrapper, log_name);
+        free(log_name);
+    }
+
+    if(m->enable_list_files)
+    {
+        log_name = string_format("%s.files", m->log_prefix);
+		makeflow_wrapper_add_output_file(m->wrapper, log_name);
+        free(log_name);
+    }
+}
+
+/*
+ * Creates a wrapper command with the appropriate resource monitor string for a given node.
+ * Returns a newly allocated string that must be freed.
+ * */
+
+char *makeflow_rmonitor_wrapper_command( struct makeflow_monitor *m, struct dag_node *n )
+{
+	char *executable;
+	if(m->exe_remote && !n->local_job){
+		executable = string_format("./%s", m->exe_remote);
```

Comment:
I wonder if it is better to just have "%s", and add the "./" to exe_remote. Right now this function changes the namespace of exe_remote, which might be undesirable.